engine/updategraph/src/main/java/io/deephaven/engine/updategraph/UpdateGraphProcessor.java

```diff
@@ -1259,6 +1259,13 @@ private interface NotificationProcessor {
          * Called before pending notifications are drained.
          */
         void beforeNotificationsDrained();
+
+        /**
+         * @return true if this processor is healthy and in an expected state
+         */
+        default boolean isHealthy() {
```

Comment:
We said we would take this out of the API, and move the assertion to `doWork()` post-acquire for the concurrent NP.